src/odemis/util/raster.py

```diff
@@ -142,3 +142,45 @@ def point_in_polygon(p, polygon):
         p1x, p1y = p2x, p2y
 
     return inside
+
+
+def get_possible_intersections(row_pairs: List[Tuple], col_pairs: List[Tuple], include_neighbors: bool = True) -> Set[Tuple]:
+    """
+    Uses Bresenham's line algorithm to determine the possible intersected fields/tiles.
```

Comment:
This needs a much clearer explanation. Please think of someone looking at this file without context. When would you want to call this function? Even knowing the context, I don't understand what it does! What's an "intersection"? What's a field/tile? Looking at the caller, it seems related to polygons? If so, why not receiving as input a "polygon" as a series of x,y coordinates?
Looking at the algorithm in detail, it seems that it determines the "pixels" that are "drawn" by drawing the series of segments passed as input (in a very odd format). Is that right?


On the other hand, the fact it uses Bresenham's line algorithm is an implementation detail. It can be worth mentioning if it ensures it's fast, or that it uses less memory, but that's a side note after explaining what it does.